core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -69,6 +70,10 @@ open class TileInfo {
 
     var naturalWonder: String? = null
     var resource: String? = null
+        set(value) {
+            tileResourceCache = null
```

Comment:
I'm usually against side effects of setters but in this case this is an elegant solution to update caches

Reply:
Academics have a name for the pattern, right?